libpimeval/src/pimPerfEnergyModels.cpp

```diff
@@ -326,8 +324,7 @@ pimParamsPerf::getPerfEnergyForFunc2(PimCmdEnum cmdType, const pimObjInfo& obj)
        * As a result, only one read operation and one write operation is necessary for the entire pass.
       */
       msRuntime = m_tR + m_tW + (maxElementsPerRegion * numberOfALUOperationPerElement * m_fulcrumAluLatency * 2) * numPass;
-      mjEnergy = obj.getNumCoresUsed() * numPass * ((maxElementsPerRegion - 1) * 3 *  m_fulcrumShiftEnergy) + ((maxElementsPerRegion) * m_fulcrumALUArithmeticEnergy * numberOfALUOperationPerElement);
-      mjEnergy += m_eAP * 3 * m_numChipsPerRank * numRanks;
+      mjEnergy = numCoresUsed * numPass * ((m_eAP * 3) + ((maxElementsPerRegion - 1) * 3 *  m_fulcrumShiftEnergy) + ((maxElementsPerRegion) * m_fulcrumALUArithmeticEnergy * numberOfALUOperationPerElement));
       mjEnergy += m_pBChip * m_numChipsPerRank * numRanks * msRuntime;
```

Comment:
The left side code is the change I made last time, and the reason I did that is because I was trying to avoid multiplying m_eAP with numCore, and only multiply it by the m_numChipsPerRank * numRanks, now it seems like I should have wrote mjEnergy += numPass * m_eAP * 3 * m_numChipsPerRank * numRanks; in line 330.

But since you are already getting a reasonable number for fulcrum, I think it would be better to just keep the changes you did! (which is a very worst case senario)